packages/mdl-auto-init/README.md

```diff
@@ -0,0 +1,130 @@
+```
+
+```html
+<div class="mdl-textfield" data-mdl-auto-init="My amazing text field!!!">
+  <!-- ... -->
+</div>
+<script>window.mdl.autoInit();</script>
+```
+
+### De-registering components
+
+Any component can be deregistered by calling `mdlAutoInit.deregister` with the name used to register
+the component.
+
+```js
+mdlAutoInit.deregister('MDLTextfield');
+```
+
+This will simply remove the name -> component mapping. It will _not_ affect any already-instantiated
+components on the page.
+
+To unregister all name -> component mappings, you can use `mdlAutoInit.deregisterAll()`.
+
+## How mdl-auto-init works
+
+`mdl-auto-init` maintains a registry object which maps string identifiers, or **names**, to
+component constructors. When the default exported function - `mdlAutoInit()` - is called,
+`mdl-auto-init` queries the DOM for all elements with a `data-mdl-auto-init` attribute. For Each
+element returned, the following steps are taken:
+
+1. If the `data-mdl-auto-init` attribute does not have a value associated with it, throw an error
+2. If the value of `data-mdl-auto-init` cannot be found in the registry, throw an error
+3. If the element has an existing property whose name is the value of `data-mdl-auto-init`, it is
+   assumed to have already been initialized. Therefore it is skipped, and a warning will be logged
+   to the console (this behavior can be overridden).
+4. Let `Ctor` be the component constructor associated with the given name in the register
+5. Let `instance` be the result of calling `Ctor.attachTo()` and passing in the element as an
+   argument.
+6. Create a non-writable, non-enumerable property on the node whose name is the value of
+   `data-mdl-auto-init` and whose value is `instance`.
+
+### Initializing only a certain part of the page
+
+By default, `mdl-auto-init` will query the entire document to figure out which components to
+initialize. To override this behavior, you can pass in an optional `root` first argument specifying
+the root node whose children will be queried for instantiation.
+
+```html
+<div id="mdl-section">
+  <!-- MDL Components, etc. -->
+</div>
+<script>window.mdl.autoInit(document.getElementById('mdl-section'));</script>
+```
+
+In the above example, only elements within `<div id="mdl-section">` will be queried.
+
+### Calling autoInit() multiple times
+
+By default, `mdl-auto-init` only expects to be called once, at page-load time. However, there may be
+certain scenarios where one may want to use `mdl-auto-init` and may still need to call it multiple
+times, such as on a Wordpress site that contains an infinitely-scrolling list of new blog post
+elements containing MDL components. `mdlAutoInit()` takes an optional second argument which is the
+function used to warn users when a component is initialized multiple times. By default, this is just
+`console.warn()`. However, to simply skip over already-initialized components without logging a
```

Comment:
Remove the first "simply" or replace it with something else, to avoid repetition.